### PR TITLE
Support for releasing M and RC releases from a tag

### PR DESCRIPTION
**Pull request by cdupuis**
This allows us to add a tag of form `1.0.0-M.1` or `1.0.0-RC.1` and have that version being used for the release npm, docker, docs and changelog goals.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,9 +22,9 @@
     "@atomist/automation-client-ext-dashboard": "0.1.0-20180705210657",
     "@atomist/automation-client-ext-logzio": "0.1.3-20180808192153",
     "@atomist/automation-client-ext-raven": "0.1.0-20180613131132",
-    "@atomist/sdm": "0.4.2-20180814004316",
-    "@atomist/sdm-core": "0.4.2-20180809202807",
-    "@atomist/sdm-pack-changelog": "0.1.2-20180814081217",
+    "@atomist/sdm": "0.4.2-20180817205802",
+    "@atomist/sdm-core": "0.4.2-20180817205826",
```
Comment (ddgenome) [inline package.json]: You should be able to use the released versions. 0.4.2 for both.

File: src/machine/release.ts [hunk 2]
```diff
@@ -82,6 +83,32 @@ async function rwlcVersion(gi: GoalInvocation): Promise<string> {
...
+    const prVersion = preReleaseVersion(gi);
+    if (prVersion) {
+        return prVersion;
+    } else {
+        return releaseVersion(version);
+    }
+}
+
+function preReleaseVersion(gi: GoalInvocation): string | undefined {
+    if (gi.sdmGoal.push.after.tags) {
```
Comment (ddgenome) [inline src/machine/release.ts]: push and after will definitely be true?

Comment (cdupuis) [reply src/machine/release.ts]: Yes. if not we are in deep trouble elsewhere in the framework.

File: package.json (hunk 1 above)
Comment (cdupuis) [reply package.json]: Ok. Although we have to do that again for the actual 1.0.0-M.1 release.

**Comment by cdupuis** [comment]
Added some tests. Strangely I couldn't find any tests for the release goal implementations at all.

**Comment by ddgenome** [comment]
I thought there were tests for some of the release stuff.

**Pull request merged**